### PR TITLE
Porting old Umbraco API Controller [Unit Testing]

### PR DESCRIPTION
**Pull request by erikjanwestendorp**
## Description

This PR belongs to: https://github.com/umbraco/UmbracoDocs/issues/6466. 

* [x] Change UmbracoApiController to a default Controller
* [x] Added route attributes
* [x] Delete obsolete warning

## Type of suggestion

* [ ] Typo/grammar fix
* [x] Updated outdated content
* [ ] New content
* [ ] Updates related to a new version
* [ ] Other

## Product & version (if relevant)
Umbraco CMS v14 & 15

### CODE REVIEW

File: 14/umbraco-cms/implementation/unit-testing.md [hunk 1]
```diff
@@ -159,23 +159,21 @@ public class PageSurfaceControllerTests
...
 See [Reference documentation on UmbracoApiControllers](../reference/routing/umbraco-api-controllers/README.md#locally-declared-controller).
 
-{% hint style="warning" %}
-The example below uses UmbracoApiController which is obsolete in Umbraco 14 and will be removed in Umbraco 15.
-{% endhint %}
-
 ```csharp
+[ApiController]
+[Route("/umbraco/api/products")]
 public class ProductsController : UmbracoApiController
```
Comment (sofietoft) [inline 14/umbraco-cms/implementation/unit-testing.md:169]: This should be changed to Controller as well, right? 🤔 

```suggestion
public class ProductsController : Controller
```

**Review by sofietoft — COMMENTED** [review]
Thanks for the PR @erikjanwestendorp ! 💪 
Just a single comment from my side.

File: 14/umbraco-cms/implementation/unit-testing.md (hunk 1 above)
Comment (erikjanwestendorp) [reply 14/umbraco-cms/implementation/unit-testing.md:169]: @sofietoft oops.. 🙈

File: 14/umbraco-cms/implementation/unit-testing.md [hunk 2]
```diff
@@ -159,23 +159,21 @@ public class PageSurfaceControllerTests
 `ServiceContext.CreatePartial()` has optional parameters, and by naming them you only need to mock the dependencies that you need, for example: `ServiceContext.CreatePartial(contentService: Mock.Of<IContentService>());`
 {% endhint %}
 
-## Testing an UmbracoApiController
+## Testing an ApiController
 
 See [Reference documentation on UmbracoApiControllers](../reference/routing/umbraco-api-controllers/README.md#locally-declared-controller).
```
Comment (sofietoft) [inline 14/umbraco-cms/implementation/unit-testing.md:164]: Should we change UmbracoApiControllers to ApiControllers here? 🤔

**Review by sofietoft — COMMENTED** [review]
I've had this past one of our developers, and it looks like the code in the test is using some legacy methods from the NUnit framework.
So, a few changes should be made here:

Line 200: Swap `AreEqual` with `That`
```
Assert.That(expected == result);
```

Line 206: Change to `JsonSerializer`
```
var json = JsonSerializer.Serialize(this.controller.GetAllProductsJson().Value);
```

Line 208: Swap `AreEqual` with `That`
```
Assert.That("[\"Table\",\"Chair\",\"Desk\",\"Computer\",\"Beer fridge\"]" == json);
```

I know this isn't related to what you're actually changing in this PR, but let's get this test fixed, then I'll create some issues for going through the rest of the tests, as they might also be using the legacy methods 💪 

Hope it all makes sense 🤞

File: 14/umbraco-cms/implementation/unit-testing.md (hunk 2 above)
Comment (erikjanwestendorp) [reply 14/umbraco-cms/implementation/unit-testing.md:164]: @sofietoft Not sure. Maybe it's better to change it to Controller?

Comment (sofietoft) [reply 14/umbraco-cms/implementation/unit-testing.md:164]: Yes, sure! I think Controller is good 💪 

Did you see my comments [above as well](https://github.com/umbraco/UmbracoDocs/pull/6524#pullrequestreview-2356241504)? 🤞

**Comment by erikjanwestendorp** [comment]
@sofietoft Thanks for the review just updated the PR with your suggestions (including the unit test) 😄

**Comment by sofietoft** [comment]
Looks great @erikjanwestendorp ! 👏 
Thanks for applying the suggestions.

I'll make sure this is merged!

**Pull request merged**